meson_options.txt

```diff
@@ -2,13 +2,11 @@ option('build', type : 'combo', choices : ['all', 'standalone', 'library'], valu
 option('agent', type : 'boolean', value : true, description : 'enable the fwupd agent')
 option('consolekit', type : 'boolean', value : true, description : 'enable ConsoleKit support')
 option('firmware-packager', type : 'boolean', value : true, description : 'enable firmware-packager installation')
-option('gpg', type : 'boolean', value : true, description : 'enable the GPG verification support')
 option('gtkdoc', type : 'boolean', value : false, description : 'enable developer documentation')
 option('introspection', type : 'boolean', value : true, description : 'generate GObject Introspection data')
 option('lvfs', type : 'boolean', value : true, description : 'enable LVFS remotes')
 option('man', type : 'boolean', value : true, description : 'enable man pages')
 option('gudev', type : 'boolean', value : true, description : 'enable GUdev support')
-option('pkcs7', type : 'boolean', value : true, description : 'enable the PKCS7 verification support')
```

Comment:
So right now there is actually the ability to compile the stack without both GPG and PKCS7 support.  IIRC this is what Google actually does when they compile into ChromeOS.  I think by moving to jcat and making it a hard dep that means it's impossible to decouple the trust verification.

@djcampello would you be able to provide some comments on your thoughts here?  If `-ENOCONTEXT`, go and take a look at https://blogs.gnome.org/hughsie/2020/02/28/introducing-jcat/

Reply:
You can compile `jcat` without GPG an without PKCS, although I'll admit I've not actually tested if fwupd still works in that case. I guess it should do.

Reply:
we could certainly add a specific CI unit test for that scenario to make sure nothing blows up when no trust libraries in place.

Reply:
Ideally I'd like something that matches what Google are doing with ChromeOS 100%...

Reply:
I don't know how realistic that actually is, they could potentially change `USE` flags from one board to another leading to differently built utilities and stack.  So I think a simple CI test that builds jcat as a subproject to fwupd but sets gpg/pkcs7 to disabled and runs the fwupd battery of tests on it would be what checks this best.

Reply:
> So right now there is actually the ability to compile the stack without both GPG and PKCS7 support. IIRC this is what Google actually does when they compile into ChromeOS. I think by moving to jcat and making it a hard dep that means it's impossible to decouple the trust verification.
> 
> @djcampello would you be able to provide some comments on your thoughts here? If `-ENOCONTEXT`, go and take a look at https://blogs.gnome.org/hughsie/2020/02/28/introducing-jcat/

Here is our current ebuild:
https://chromium.googlesource.com/chromiumos/overlays/chromiumos-overlay/+/master/sys-apps/fwupd/fwupd-9999.ebuild

It seems that at least we have IUSE=+gpg

Reply:
@djcampello I'm not super familiar with ebuilds, with `IUSE=+gpg` does that mean that by default the ebuild has gpg turned off, and a board would need to explicitly build with `-gpg` to not have it?

Can you confirm then for example that a sarien build ended up with gpg turned on in the end?

Reply:
Actually I was mistaken... Even though is IUSE=+gpg... gpg is not used... Here are the actual USE flags on sarien:

sys-apps/fwupd-9999::chromiumos was built with the following:
USE="minimal nvme -agent -amt -consolekit -cros_host -dell -elogind -gpg -gtk-doc -introspection -man -nls -pkcs7 -redfish -synaptics -systemd -test -thunderbolt -uefi"

None of gpg or pkcs7 are used

Reply:
Well I experimented with trying to build a docker container that was basically this:
* Start out with gentoo stage3 amd64
* Pull in portage
* Clone chromiumos-overlay, eclass-overlay
* emerge fwupd-9999

But I kept getting caught up on stuff that was very chromium specific (references to portage-stable, boards etc).  I think if we're going to have a proper CI test that builds it exactly "like ChromeOS does" to catch this stuff early we'll need you to help put it together.

For now I pushed something into this branch (`contrib/ci/trust.sh`) that will build with at least the flags I gathered should be matching what you shared above with pkcs7 and gpg turned off.